xchange-bitstamp/src/main/java/org/knowm/xchange/bitstamp/BitstampAdapters.java

```diff
@@ -300,80 +301,24 @@ public static UserTrades adaptTradeHistory(BitstampUserTransaction[] bitstampUse
     return fundingRecords;
   }
 
-  private static CurrencyPair adaptCurrencyPair(BitstampOrderTransaction transaction) {
-
-    // USD section
-    if (transaction.getBtc() != null && transaction.getUsd() != null) return CurrencyPair.BTC_USD;
-
-    if (transaction.getLtc() != null && transaction.getUsd() != null) return CurrencyPair.LTC_USD;
-
-    if (transaction.getEth() != null && transaction.getUsd() != null) return CurrencyPair.ETH_USD;
-
-    if (transaction.getXrp() != null && transaction.getUsd() != null) return CurrencyPair.XRP_USD;
-
-    if (transaction.getBch() != null && transaction.getUsd() != null) return CurrencyPair.BCH_USD;
-
-    // EUR section
-    if (transaction.getBtc() != null && transaction.getEur() != null) return CurrencyPair.BTC_EUR;
-
-    if (transaction.getLtc() != null && transaction.getEur() != null) return CurrencyPair.LTC_EUR;
-
-    if (transaction.getEth() != null && transaction.getEur() != null) return CurrencyPair.ETH_EUR;
-
-    if (transaction.getXrp() != null && transaction.getEur() != null) return CurrencyPair.XRP_EUR;
-
-    if (transaction.getBch() != null && transaction.getEur() != null) return CurrencyPair.BCH_EUR;
-
-    // BTC section
-    if (transaction.getLtc() != null && transaction.getBtc() != null) return CurrencyPair.LTC_BTC;
-
-    if (transaction.getEth() != null && transaction.getBtc() != null) return CurrencyPair.ETH_BTC;
-
-    if (transaction.getXrp() != null && transaction.getBtc() != null) return CurrencyPair.XRP_BTC;
-
-    if (transaction.getBch() != null && transaction.getBtc() != null) return CurrencyPair.BCH_BTC;
-
-    // XLM
-    if (transaction.getXlm() != null && transaction.getBtc() != null) return CurrencyPair.XLM_BTC;
-
-    if (transaction.getXlm() != null && transaction.getUsd() != null) return CurrencyPair.XLM_USD;
+  private static CurrencyPair adaptCurrencyPair(BitstampOrderTransaction transaction, List<CurrencyPair> exchangeSymbols) {
 
-    if (transaction.getXlm() != null && transaction.getEur() != null) return CurrencyPair.XLM_EUR;
-
-    if (transaction.getXlm() != null && transaction.getEth() != null) return CurrencyPair.XLM_ETH;
-
-    // LINK
-    if (transaction.getLink() != null && transaction.getBtc() != null) return CurrencyPair.LINK_BTC;
-
-    if (transaction.getLink() != null && transaction.getUsd() != null) return CurrencyPair.LINK_USD;
-
-    if (transaction.getLink() != null && transaction.getEur() != null) return CurrencyPair.LINK_EUR;
-
-    if (transaction.getLink() != null && transaction.getEth() != null) return CurrencyPair.LINK_ETH;
+    String[] keys = transaction.getAmounts().keySet().toArray(new String[0]);
+    if (keys.length != 2) {
+      throw new IllegalArgumentException("Amount size is not 2. Unable to calculate currency pair.");
+    }
 
-    throw new NotYetImplementedForExchangeException();
+    if (exchangeSymbols.contains(new CurrencyPair(keys[0], keys[1]))) {
+      return new CurrencyPair(keys[0], keys[1]);
```

Comment:
minor memory optimization, rather then return `new CurrencyPair(keys[0], keys[1])` can you just return `exchangeSymbols.get(new CurrencyPair(keys[0], keys[1]))`

Reply:
List.get() does not accept object but index. But I have created a variable for the same purpose.